extensions/sql-database-projects/src/models/publishProfile/publishProfile.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as xmldom from 'xmldom';
+import * as constants from '../../common/constants';
+import * as utils from '../../common/utils';
+
+import { promises as fs } from 'fs';
+import { Uri } from 'vscode';
+import { SqlConnectionDataSource } from '../dataSources/sqlConnectionStringSource';
+import { ApiWrapper } from '../../common/apiWrapper';
+
+
+// only reading db name, connection string, and SQLCMD vars from profile for now
+export interface PublishProfile {
+	databaseName: string;
+	connectionId: string;
+	connectionString: string;
+	sqlCmdVariables: Record<string, string>;
+}
+
+/**
+ * parses the specified file to load publish settings
+ */
+export async function load(profileUri: Uri, apiWrapper: ApiWrapper,): Promise<PublishProfile> {
```

Comment:
any secific reason for not having publish profile as class and adding object intances in project but global function?

Reply:
I was following how DataSources was implemented for having this load function be an exported function. No reason for it being an interface. Would it make more sense for it to be a class?